app/__init__.py

```diff
@@ -790,6 +791,37 @@ def add_review():
     sd_data = ShoreDivingData.query.filter_by(id=sd_id).first_or_404()
     beach_id = sd_data.spot_id
 
+  spot = Spot.query.filter_by(id=beach_id).first()
+  if not spot:
+    return { 'msg': 'Couldn\'t find that spot' }, 404
+
+  if len(buddies) > 0:
+    if activity_type == 'snorkel':
+      activity = 'snorkeled'
+    else:
+      activity = 'dived'
+    if len(buddies) == 1:
+      text += (f" I {activity} with {len(buddies)} other buddy.")
+    else:  
+      text += (f" I {activity} with {len(buddies)} other buddies.")
+    
+    buddy_message = Mail(
+      from_email=('hello@zentacle.com', 'Zentacle'),
+      to_emails=buddies)
+
+    buddy_message.template_id = 'd-8869844be6034dd09f0d7cc27e27aa8c'
+    buddy_message.dynamic_template_data = {
+      'display_name': user.username ,
```

Comment:
this should be user.display_name

Reply:
ah, that would explain why it wasn't reading correctly.